framework/src/modules/http_api/helpers/filter_peers.js

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const _ = require('lodash');
+
+/**
+ * Sorts peers.
+ *
+ * @todo Add @param tags
+ * @todo Add @returns tag
+ * @todo Add description of the function
+ */
+
+const sortPeers = function(field, asc) {
+	return function(a, b) {
+		// Match the default JavaScript sort order.
+		if (a[field] === b[field]) {
+			return 0;
+		}
+		// Ascending
+		if (asc) {
+			// Undefined last
+			if (a[field] === undefined) {
+				return 1;
+			}
+			if (b[field] === undefined) {
+				return -1;
+			}
+			// Null second last
+			if (a[field] === null) {
+				return 1;
+			}
+			if (b[field] === null) {
+				return -1;
+			}
+			if (a[field] < b[field]) {
+				return -1;
+			}
+			return 1;
+		}
+		// Descending
+		// Undefined first
+		if (a[field] === undefined) {
+			return -1;
+		}
+		if (b[field] === undefined) {
+			return 1;
+		}
+		// Null second
+		if (a[field] === null) {
+			return -1;
+		}
+		if (b[field] === null) {
+			return 1;
+		}
+		if (a[field] < b[field]) {
+			return 1;
+		}
+		return -1;
+	};
+};
+
+/**
+ * Returns peers length by filter but without offset and limit.
+ * @param {Array} peers
+ * @param {Object} filter
+ * @returns {int} count
+ * @todo Add description for the params
+ */
+const getByFilter = function(peers, filter) {
+	const allowedFields = [
+		'ip',
+		'wsPort',
+		'httpPort',
+		'state',
+		'os',
+		'version',
+		'protocolVersion',
+		'broadhash',
+		'height',
+		'nonce',
+	];
+	const limit = filter.limit ? Math.abs(filter.limit) : null;
```

Comment:
I think limit has the default of 20 and max 100?

Reply:
Actually, this function is used for two things, one to get all the peer based on all the filters and other is the number of count of peers for those filters without taking `limit` and `offset` into account. Also, http_api is already taking care of the default value for `limit` as 10 and `offset` as 0.